### ISSUE TITLE
Openstack failed job should retry on a different host

### ISSUE THREAD

**Issue by keiranmraine** [VertebrateResequencing/wr#72]
If a jobs reties it generally is going to end up of the same host if there is nothing else in the queue (not sure if retry behaviour is to mark job as high priority).  If it is a requirement to run on a different host in many cases the original will be destroyed.

**Comment by sb10** [comment]
I could implement this as "if using a cloud scheduler, if `wr runner` attempts to run a command and it exits non-zero, die instead of picking up a new command from the queue".

It is already implemented that if a runner dies, the host is immediately terminated.

Would this work for you? I do not see this as a user option. It would always happen.


... Alternatively I could implement it for all schedulers as "if `wr runner` attempts to run a command and it exits non-zero, and if there are no more pending command in the queue to run, exit instead of allowing this failed command to come back to us".

**Comment by keiranmraine** [comment]
Yes I think this would work.  We'd just have to ensure any post job clean up passes the correct exit code through (we capture logs, push them to S3 and cleanup disk on failures).

**Comment by sb10** [comment]
Would you be able to put your code for capturing logs and pushing them to s3 in a "run" behaviour of https://github.com/VertebrateResequencing/wr/issues/71 , and use the cleanup feature? Do we even need a "copy_to_manager" behaviour, since you can just stick them in S3?

This way the exit code is for your "real" command, not for your own cleanup code.
